packages/markdown-support/src/types.ts

```diff
@@ -1,6 +1,16 @@
+import unified from 'unified';
+
 export interface AstroMarkdownOptions {
   /** Enable or disable footnotes syntax extension */
   footnotes: boolean;
   /** Enable or disable GitHub-flavored Markdown syntax extension */
   gfm: boolean;
+  remarkPlugins: Promise<{ default: unified.Plugin }>[];
+  rehypePlugins: Promise<{ default: unified.Plugin }>[];
```

Comment:
I think we also want to support `[Promise<{ default: unified.Plugin }>, any]` here so plugins can accept configuration

Reply:
```suggestion
  remarkPlugins: (Promise<{ default: unified.Plugin }>|[Promise<{ default: unified.Plugin }>, any])[];
  rehypePlugins: (Promise<{ default: unified.Plugin }>|[Promise<{ default: unified.Plugin }>, any])[];
```

Reply:
Yep, I had it in mind :) Thanks for the notice!